app/controllers/locations_controller.rb

```diff
@@ -77,20 +82,20 @@ def find_scheme
     @scheme = if %w[new create index edit_name].include?(action_name)
                 Scheme.find(params[:id])
               else
-                @location.scheme
+                @location&.scheme
               end
   end
 
   def find_location
-    @location = params[:location_id].present? ? Location.find(params[:location_id]) : Location.find(params[:id])
+    @location = params[:location_id].present? ? Location.find_by(id: params[:location_id]) : Location.find_by(id: params[:id])
```

Comment:
what is the difference here? I thought find was guaranteed to return a single result based on the ID.

Reply:
`find_by` returns nil when no results. `find` throws an error.